src/main/java/it/rate/webapp/controllers/api/ImageRestController.java

```diff
@@ -33,11 +33,12 @@ public ResponseEntity<?> getImage(@PathVariable String id) {
 
   @PostMapping("/new-interest-image")
   @PreAuthorize("@permissionService.canCreateInterest()")
-  public ResponseEntity<?> uploadNewInterestImage(@RequestParam("picture") MultipartFile file) {
+  public ResponseEntity<?> uploadNewInterestImage(@RequestParam("picture") MultipartFile file, Principal principal) {
 
+    String userName = principal.getName();

```

Comment:
The username obtained here is in fact email. I would prefer to name this according to that here and in all subsequent places for clarity.